python/lsst/pipe/base/argumentParser.py

```diff
@@ -732,19 +724,28 @@ def __call__(self, parser, namespace, values, option_string):
         @param[in] parser       argument parser (instance of ArgumentParser)
         @param[in] namespace    parsed command (an instance of argparse.Namespace); ignored
         @param[in] values       a list of trace levels;
-            each item must be of the form component_name=level
+            each item must be of the form component_name=level or level
+            where level is an integer or a keyword
         @param[in] option_string    option value specified by the user (a str)
         """
+        permittedLevelList = ('DEBUG', 'INFO', 'WARN', 'FATAL')
+        permittedLevelSet = set(permittedLevelList)
         for componentLevel in values:
             component, sep, levelStr = componentLevel.partition("=")
             if not levelStr:
-                parser.error("%s level %s must be in form component=level" % (option_string, componentLevel))
-            try:
-                level = int(levelStr)
-            except Exception:
-                parser.error("cannot parse %r as an integer level for %s" % (levelStr, component))
-            pexLog.Trace.setVerbosity(component, level)
-
+                levelStr, component = component, ""
+            logLevelUpr = levelStr.upper()
+            if logLevelUpr in permittedLevelSet:
+                logLevel = getattr(namespace.log, logLevelUpr)
+            else:
+                try:
+                    logLevel = int(levelStr)
+                except Exception:
```

Comment:
I do not see the value of including the error message. In my opinion the reported error says all that the user needs to know, and I worry about adding needless clutter. What am I missing?